compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -719,6 +690,20 @@ fn encode_ty<'tcx>(
             typeid.push_str(&s);
         }
 
+        // Associated type projections
```

Comment:
Aliases are not real types. They should never be encountered during codegen.

Reply:
This is also related to the information I provided above.

Reply:
Related how?

Reply:
Sorry, I should've elaborated. As with `ReEarlyParam` and `Param`, they may also be present in the trait identity of the trait that implements the method.

Reply:
Yes, but it's not correct to be using the identity substituted trait predicates like ever. Codegen only operates on properly monomorphized code.